analysis/prep_sg.R

```diff
@@ -0,0 +1,99 @@
+####
+## This script does the following:
+# 1. Import the processed dataset of all eligible participants
+# 2. Create subsets to run subgroup analyses using the reusable action cox-ipw
+# 3. Save all subsets
+####
+
+
+# Import libraries and user functions -------------------------------------
+print('Import libraries and functions')
+library('tidyverse')
+library('here')
+library('arrow')
+
+
+# Create directories for output -------------------------------------------
+print('Create directories for output')
+fs::dir_create(here::here("output", "data", "sg"))
+
+
+# Import the data ---------------------------------------------------------
+print('Import the data')
+df <- read_feather(here("output", "data", "data_processed.arrow"))
+
+
+# Import dates ------------------------------------------------------------
+print('Import dates')
+source(here::here("analysis", "metadates.R"))
+study_dates <- lapply(study_dates, function(x) as.Date(x))
+studyend_date <- as.Date(study_dates$studyend_date, format = "%Y-%m-%d")
+
+
+# Subset the data ---------------------------------------------------------
```

Comment:
This approach creates and maintains multiple copies of the data in the environment, which is not advisable. I would recommend that you create a function where you pass the subgroup name in as a parameter. The function should do the filtering, prepare and save the dataset, and then clean up. You will then only have the full copy of the data that you are drawing from and the subgroup you are working on in the environment at any one time.